packages/engine/paima-sm/src/index.ts

```diff
@@ -384,6 +386,30 @@ async function processPaginatedCdeData(
   });
 }
 
+// Process Batcher Payments
+async function processBatcherPayments(
+  latestChainData: ChainData,
+  DBConn: PoolClient
+): Promise<number> {
+  let count = 0;
+
+  // Update batcher payments and fees.
+  for (const [query, params] of latestChainData.batcher?.sqlUpdates ?? []) {
+    count += 1;
+    await query.run(params, DBConn);
+  }
+
+  for (const e of latestChainData.batcher?.events ?? []) {
+    await PaimaEventManager.Instance.sendMessage(BuiltinEvents.BatcherPayment, {
```

Comment:
careful: it's possible for this database transaction to revert, in which case you've send events for a tx that never happened

Reply:
For the Batcher this should be expected, as gas fees should be charged as the TX was posted independent of the outcome (or we leave a loophole where a player could drain the batcher's funds by sending bad requests)

Reply:
makes sense, but we should put a comment explaining this

Reply:
added